### PR TITLE
eth2util/enr: port minimal implementation

### PR DESCRIPTION
**Pull request by corverroos**
Adds minimal ENR implementation.

category: feature
ticket: #1608

### CODE REVIEW

File: eth2util/enr/enr.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	}
+
+	return Record{
+		PubKey:    &privkey.PublicKey,
+		Signature: sig,
+	}, nil
+}
+
+// Record represents an Ethereum Node Record.
+type Record struct {
```
Comment (dB2510) [inline eth2util/enr/enr.go:106]: is this compatible with records already generated by enr.Record from geth?

**Review by dB2510 — APPROVED** [review]
LGTM

File: eth2util/enr/enr.go (hunk 1 above)
Comment (corverroos) [reply eth2util/enr/enr.go:106]: no

File: eth2util/enr/enr.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+				return Record{}, errors.New("non-v4 identity scheme not supported")
+			}
+		}
+	}
+
+	if r.PubKey == nil {
+		return Record{}, errors.New("missing secp256k1 public key")
+	}
+
+	if err := verify(r.PubKey, r.Signature, rlp.EncodeBytesList(elements[1:])); err != nil {
```
Comment (xenowits) [inline eth2util/enr/enr.go:85]: why is the first element excluded?

**Pull request merged**